src/feedback/commands/submitFeedbackListener.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { _Blob } from 'aws-sdk/clients/lambda'
+import { getLogger, Logger } from '../../shared/logger'
+import { TelemetryService } from '../../shared/telemetry/telemetryService'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+
+export interface FeedbackMessage {
+    command: string
+    comment?: string
+    sentiment?: string
+}
+
+export interface FeedbackPanel {
+    postMessage(message: any): Thenable<boolean>
+    dispose(): any
+}
+
+export interface Window {
+    showInformationMessage(message: string): Thenable<string | undefined>
+}
+
+export function submitFeedbackListener(panel: FeedbackPanel, window: Window, telemetry: TelemetryService) {
+    const logger: Logger = getLogger()
+
+    return async (message: FeedbackMessage) => {
+        switch (message.command) {
+            case 'submitFeedback':
+                logger.info(`Submitting ${message.sentiment} feedback`)
+
+                try {
+                    await telemetry.postFeedback({
```

Comment:
What happens if the user has telemetry disabled? Will the feedback still go through? If not, should we inform the user of that?

Also, will this be emitted instantaneously or do we need to wait for the telemetry cache to clear? If we have to wait for the cache to clear, there's no point in catching a server error at this level.

Reply:
Good question. 

> What happens if the user has telemetry disabled? Will the feedback still go through? If not, should we inform the user of that?

Yes, it still goes through. Normally the telemetry publisher gets created lazily the first time metrics are published. It normally doesn't get created if telemetry is disabled. This will trigger the creation, but it still won't publish metrics as the metrics flush checks the flag.

> Also, will this be emitted instantaneously or do we need to wait for the telemetry cache to clear?

Instantaneously and it returns backs Errors if the call fails which are then displayed in the web view for the user to retry.